doc/stdenv/stdenv.chapter.md

```diff
@@ -77,7 +77,7 @@ where the builder can do anything it wants, but typically starts with
 source $stdenv/setup
 ```
 
-to let `stdenv` set up the environment (e.g., process the `buildInputs`). If you want, you can still use `stdenv`’s generic builder:
+to let `stdenv` set up the environment (e.g. by resetting `PATH` and populating it from `buildInputs`). If you want, you can still use `stdenv`’s generic builder:
```

Comment:
```suggestion
to let `stdenv` set up the environment (e.g. by resetting `PATH` and populating it from `nativeBuildInputs`). If you want, you can still use `stdenv`’s generic builder:
```

Make it strictDeps-proof.

Reply:
I'm not sure what you mean. Can you point to the implementation to show what you mean?

It [resets `PATH`](https://github.com/NixOS/nixpkgs/blob/8d83fa4198313a3f0ec1d521b37ea52477e26c8c/pkgs/stdenv/generic/setup.sh#L274-L275), [possibly has some dead code](https://github.com/NixOS/nixpkgs/blob/8d83fa4198313a3f0ec1d521b37ea52477e26c8c/pkgs/stdenv/generic/setup.sh#L306-L309) in between, and then does some [deep black magic](https://github.com/NixOS/nixpkgs/blob/8d83fa4198313a3f0ec1d521b37ea52477e26c8c/pkgs/stdenv/generic/setup.sh#L471-L503) @Ericson2314 to set up all dependencies. All the `*Inputs` varibles seem to come from [here](https://github.com/NixOS/nixpkgs/blob/8d83fa4198313a3f0ec1d521b37ea52477e26c8c/pkgs/stdenv/generic/default.nix#L131-L132), but there is at least another layer around it where users of `stdenv.mkDerivation` actually specify `buildInputs` and friends, but at that point I can't follow any more.

Reply:
if you set ``strictDeps = true`` only nativeBuildInputs are added to ``$PATH`` and have the architecture of the machine the build is being run on. ``buildInputs`` then can be the same architecture or something totally different.

Reply:
This is correct, but does not seem like the right place to go into such detail.

Perhaps the original text ("process the `buildInputs`") was better here, because the usual operation of `nativeBuildInputs` is rather simple in comparison. If was writing a derivation without `setup.sh`, I'd just set `PATH` myself. `buildInputs` otoh has logic for various languages and tools that I wouldn't want to take care of manually.

Reply:
I wanted to specify what exactly is done with `buildInputs`, because the token means something specific. Since reality is messy, maybe let's use the vague prose capturing all the other things considered "build inputs"?

```suggestion
to let `stdenv` set up the environment (e.g. by processing build inputs). If you want, you can still use `stdenv`’s generic builder:
```

Reply:
sgtm!

Reply:
Looks like you forgot to apply your own suggestion. (unresolving)